core/services/ocr2/plugins/median/plugin.go

```diff
@@ -0,0 +1,82 @@
+package median
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+	"github.com/smartcontractkit/libocr/offchainreporting2/reportingplugin/median"
+	ocr2types "github.com/smartcontractkit/libocr/offchainreporting2/types"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/job"
+	"github.com/smartcontractkit/chainlink/core/services/ocr2/plugins"
+	"github.com/smartcontractkit/chainlink/core/services/ocrcommon"
+	"github.com/smartcontractkit/chainlink/core/services/pipeline"
+	"github.com/smartcontractkit/chainlink/core/services/relay/types"
+)
+
+// The Median struct holds parameters needed to run a Median plugin.
+type Median struct {
+	jobSpec        job.Job
+	ocr2Provider   types.OCR2Provider
+	pipelineRunner pipeline.Runner
+	runResults     chan pipeline.Run
+	lggr           logger.Logger
+	ocrLogger      commontypes.Logger
+
+	config PluginConfig
+}
+
+var _ plugins.OraclePlugin = Median{}
+
+// NewMedian parses the arguments and returns a new Median struct.
+func NewMedian(jobSpec job.Job, ocr2Provider types.OCR2Provider, pipelineRunner pipeline.Runner, runResults chan pipeline.Run, lggr logger.Logger, ocrLogger commontypes.Logger) (Median, error) {
```

Comment:
could we constraint scope by just passing in the jobID and the pipeline spec here? (vs giving this access to all of job.Job)

Reply:
We use the entire job in two places `ocrcommon.NewInMemoryDataSource` and `ocrcommon.NewDataSourceV2`

Reply:
ah yeah those don't really need the full job either, they just need the name, ID/externalID

Reply:
so we can pass those into the data sources as separate args. Not blocking though